docs/standard/collections/when-to-use-generic-collections.md

```diff
@@ -1,51 +1,70 @@
+
+- <xref:System.Collections.Generic.List%601> is the generic class that corresponds to <xref:System.Collections.ArrayList>.
+
+- <xref:System.Collections.Generic.Dictionary%602> and <xref:System.Collections.Concurrent.ConcurrentDictionary%602> are the generic classes that correspond to <xref:System.Collections.Hashtable>.
+
+- <xref:System.Collections.ObjectModel.Collection%601> is the generic class that corresponds to <xref:System.Collections.CollectionBase>. <xref:System.Collections.ObjectModel.Collection%601> can be used as a base class, but unlike <xref:System.Collections.CollectionBase>, it is not abstract, which makes it much easier to use.
+
+- <xref:System.Collections.ObjectModel.ReadOnlyCollection%601> is the generic class that corresponds to <xref:System.Collections.ReadOnlyCollectionBase>. <xref:System.Collections.ObjectModel.ReadOnlyCollection%601> is not abstract, and has a constructor that makes it easy to expose an existing <xref:System.Collections.Generic.List%601> as a read-only collection.
+
+- The <xref:System.Collections.Generic.Queue%601>, <xref:System.Collections.Concurrent.ConcurrentQueue%601>, <xref:System.Collections.Immutable.ImmutableQueue%601>, <xref:System.Collections.Generic.Array%601>, <xref:System.Collections.Concurrent.ConcurrentArray%601>, <xref:System.Collections.Immutable.ImmutableArray%601>, <xref:System.Collections.Generic.SortedList%602>, and <xref:System.Collections.Immutable.ImmutableSortedSet%601> generic classes correspond to the respective nongeneric classes with the same names.
+
+## Additional Types
+
+Several generic collection types do not have nongeneric counterparts. They include the following:
+
+- <xref:System.Collections.Generic.LinkedList%601> is a general-purpose linked list that provides O(1) insertion and removal operations.
+
+- <xref:System.Collections.Generic.SortedDictionary%602> is a sorted dictionary with O(log `n`) insertion and retrieval operations, which makes it a useful alternative to <xref:System.Collections.Generic.SortedList%602>.
+
+- <xref:System.Collections.ObjectModel.KeyedCollection%602> is a hybrid between a list and a dictionary, which provides a way to store objects that contain their own keys.
+
+- <xref:System.Collections.Concurrent.BlockingCollection%601> implements a collection class with bounding and blocking functionality.
+
+- <xref:System.Collections.Concurrent.ConcurrentBag%601> provides fast insertion and removal of unordered elements.
+
+### Immutable builders
+
+When considering immutability as a functional desire of application development, the <xref:System.Collections.Immutable> namespace offers generic collection types. All of the immutable collection types offer `Builders` as an optimization to performing numerous mutations. `Builders` work to batch operations in an unfrozen state. When all mutations have been completed on a `Builder` instance, call `ToImmutable` to freeze all nodes and create an `Immutable*<T>` generic collection.
```

Comment:
```suggestion
When you desire immutability functionality in your app, the <xref:System.Collections.Immutable> namespace offers generic collection types you can use. All of the immutable collection types offer `Builder` subclasses that can optimized performance when you're performing multiple mutations. The `Builder` class batches operations in a mutable state. When all mutations have been completed, call the `ToImmutable` method to "freeze" all nodes and create an immutable generic collection, for example, an <xref:System.Collections.Immutable.ImmutableList%601>.
```

In looking at the preview for this page, I think there's probably too much duplication in the bullets that follow. I suggest listing all of the builder subclasses (is that the right term?) and then just describing the process of which methods to call for _one_ of those types.

Reply:
I agree and I'll update that tomorrow in a separate PR.